File: cont/base/springcontent/shaders/GLSL/Particles/FlameParticleGenerator.lua
return {
	MainCode =
[[
	float currTime = frameInfo.x + frameInfo.y - createFrame;

	SetCurrentAnimation(animParams, currTime);
	vec3 drawPos = pos + speed * frameInfo.y;
	float colMixRate = clamp((colEdge1 - curTime)/(colEdge1 - colEdge0), 0.0, 1.0);
	vec4 color = mix(color0, color1, colMixRate);

	AddEffectsQuadCamera(
		drawOrder,
		animParams,
		drawPos, vec2(size), texCoord,
		color
	);
]]
}

File: rts/Rendering/Env/Particles/Generators/FlameParticleGenerator.cpp
#include "FlameParticleGenerator.h"

CR_BIND(FlameParticleData, )
CR_REG_METADATA(FlameParticleData,
(
	CR_MEMBER(pos),
	CR_MEMBER(drawRadius),
	CR_MEMBER(speed),
	CR_MEMBER(createFrame),
	CR_MEMBER(animParams),
	CR_MEMBER(drawOrder),
	CR_MEMBER(rotParams),
	CR_MEMBER(curTime),
	CR_MEMBER(color0),
	CR_MEMBER(color1),
	CR_MEMBER(colEdge0),
	CR_MEMBER(colEdge1),
	CR_MEMBER(texCoord)
))